app/src/main/java/com/keylesspalace/tusky/components/viewthread/ViewThreadViewModel.kt

```diff
@@ -144,8 +139,11 @@ class ViewThreadViewModel @Inject constructor(
             // for the status. Ignore errors, the user still has a functioning UI if the fetch
             // failed.
             if (timelineStatus != null) {
-                val viewData = api.status(id).getOrNull()?.toViewData(isDetailed = true)
-                if (viewData != null) { detailedStatus = viewData }
+                val result = api.status(id).getOrNull()
+                if (result != null) {
+                    eventHub.dispatch(StatusChangedEvent(result))
```

Comment:
This is not ideal since `ViewThreadViewModel` is both sending an observing `StatusChangedEvent`, but I don't have a better idea :/

Reply:
I wondered a bit why this case here stands out.

And maybe it doesn't: What happens if I refresh a timeline and one of the status there did change?
(A status that might as well be shown on the bookmark timeline for example?)

Reply:
Yes maybe we should send the event for all statuses 🤔 
Still there is a problem that this ViewModel is both sending and receiving

Reply:
Just a few ideas from thinking about this (and maybe questions). The concept(s) are still blurry in my mind:

If this "UpdateEvent" informs about updates (only) then there should be an actual check if there was an update.
That in turn needs a sort of central instance which stores every status that every activity in the running instance has seen. Only this instance/thing/repository would compare and if needed send a changed event.

Otherwise one would need quite a few locations which would need to send every status they see as "change", because they _might_ have been changed somewhere else?

(Is that actually a wide-spread problem: activities remaining active even when another one is in the foreground and then their ui must be refreshed for every change/data received everywhere else?)

And specifically for the thread view: It receives the event to update its ui (on these status)?
Maybe it should then receive _all_ the data to display as event - esp. the events it sends itself?
(It sounds too complex when it should simply "load things and then display them" but maybe that is because of the discussion above).

Sorry for writing so long but maybe it helps a bit.

Reply:
> This is not ideal since `ViewThreadViewModel` is both sending and observing `StatusChangedEvent`, but I don't have a better idea :/

https://developer.android.com/topic/architecture/data-layer

Tell the data layer to update the data, and the data layer content should be exposed as a `Flow`, so that consumers of the content observe changes as they happen.

See e.g., https://developer.android.com/codelabs/android-room-with-a-view-kotlin#6, https://developer.android.com/topic/architecture/data-layer/offline-first#reads, https://developer.android.com/topic/architecture/data-layer/offline-first#writes

That'll be a lot easier when https://github.com/tuskyapp/Tusky/pull/3436 is complete.

Reply:
I kinda agree that observing the DB is kind of the best thing but also what's the problem with both sending and receiving?

Reply:
It's a layering violation, and a source of bugs. With this approach every view model has to emit events, and if that's missed, or one of them has the wrong logic, then you have a bug.

By putting it in the data layer you guarantee that it doesn't matter what is updating the data, every update will generate the correct event.

Reply:
I agree with this, model and persistence should be the source of truth. I was referring to @connyduck 's comment what's the problem with both sending and receiving the event

Reply:
Yes basically what Nik says. 
Problem is, we don't have a data layer for non-cached timelines so I'll fix it for cached ones only for now, still better than before.

Reply:
That's coming in https://github.com/tuskyapp/Tusky/pull/3436. I'm in the "Cleanup up the TODOs and update the tests" phase in that PR.